spec/syntax/parser_spec.cr

```diff
@@ -2474,8 +2474,29 @@ describe "Parser" do
     fn
       ->(1, :hi) { true || false }
     end
-  ),                        AnonymousFunction.new([Block.new([p(nil, l(1)), p(nil, l(:hi))], e(Or.new(l(true), l(false))))])
+  ),                          AnonymousFunction.new([Block.new([p(nil, l(1)), p(nil, l(:hi))], e(Or.new(l(true), l(false))))])
 
+  # Exception handling is not allowed with {... } syntax, but only with do... end
+  it_does_not_parse %q(
+    fn
+      ->() {
+        raise :error
+        rescue
+          :rescued
+      }
+    end
+  )
+
+  # Expection handling in anonymous functions with do... end syntax is allowed
+  it_parses %q(
+    fn
+      ->() do
+        raise :error
+        rescue
```

Comment:
`rescue` and `end` here should be indented to match the `->()` that starts the clause (similar to how they match with `def` or other blocks).